### PR TITLE
Support for DPS enrollment groups

### PR DESCRIPTION
**Pull request by CIPop**
Updating PR #1450 to address comments as the original branch was deleted by the author.

1. Addressing PR feedback.
2. Correcting DPS Primary Key to match documentation.
3. Adding the same message in the LL sample.

### CODE REVIEW

File: provisioning_client/samples/prov_dev_client_ll_sample/prov_dev_client_ll_sample.c [hunk 1]
```diff
@@ -163,6 +163,8 @@ int main()
     (void)IoTHub_Init();
     (void)prov_dev_security_init(hsm_type);
     // Set the symmetric key if using they auth type
+    // If using DPS with an enrollment group, this must the the derived device key from the DPS Primary Key
+    // https://docs.microsoft.com/en-us/azure/iot-dps/concepts-symmetric-key-attestation?tabs=azure-cli#group-enrollments
```
Comment (danewalton) [inline provisioning_client/samples/prov_dev_client_ll_sample/prov_dev_client_ll_sample.c:167]:
```suggestion
    // https://docs.microsoft.com/azure/iot-dps/concepts-symmetric-key-attestation?tabs=azure-cli#group-enrollments
```

File: provisioning_client/samples/prov_dev_client_sample/prov_dev_client_sample.c [hunk 2]
```diff
@@ -103,8 +103,8 @@ int main()
     (void)prov_dev_security_init(hsm_type);
 
     // Set the symmetric key if using they auth type
-    //If using DPS with an enrollment group, this must the the derived key from the DPS Master key
-    //https://docs.microsoft.com/en-us/azure/iot-dps/how-to-legacy-device-symm-key#derive-a-device-key
+    // If using DPS with an enrollment group, this must the the derived device key from the DPS Primary Key
+    // https://docs.microsoft.com/en-us/azure/iot-dps/concepts-symmetric-key-attestation?tabs=azure-cli#group-enrollments
```
Comment (danewalton) [inline provisioning_client/samples/prov_dev_client_sample/prov_dev_client_sample.c:107]:
```suggestion
    // https://docs.microsoft.com/azure/iot-dps/concepts-symmetric-key-attestation?tabs=azure-cli#group-enrollments
```

File: provisioning_client/samples/prov_dev_client_ll_sample/prov_dev_client_ll_sample.c (hunk 1 above)
Comment (danewalton) [reply provisioning_client/samples/prov_dev_client_ll_sample/prov_dev_client_ll_sample.c:167]: allows for redirect to other langauges

**Review by danewalton — APPROVED** [review]

**Comment by CIPop** [comment]
Thanks @danewalton!

**Pull request merged**